src/reconcile/rec_col.c

```diff
@@ -760,17 +760,7 @@ __wt_rec_col_var(
             if (ins != NULL && WT_INSERT_RECNO(ins) == src_recno) {
                 WT_ERR(__wt_rec_upd_select(session, r, ins, cip, vpack, &upd_select));
                 upd = upd_select.upd;
-                if (upd == NULL) {
```

Comment:
Here upd equals null means we are writing the same value to disk thus the time pairs should be unchanged.

Reply:
My understanding here is wrong. This is in the insert list so we are also handling new keys here.